runtime/js/40_command.js

```diff
@@ -0,0 +1,191 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+"use strict";
+
+((window) => {
+  const core = window.Deno.core;
+  const { pathFromURL } = window.__bootstrap.util;
+  const { read, write } = window.__bootstrap.io;
+  const { illegalConstructorKey } = window.__bootstrap.webUtil;
+  const { ArrayPrototypeMap, ObjectEntries, String, TypeError } =
+    window.__bootstrap.primordials;
+
+  function createWritableIOStream(rid) {
+    return new WritableStream({
+      async write(chunk) {
+        let nwritten = 0;
+        while (nwritten < chunk.byteLength) {
+          nwritten += await write(rid, chunk.subarray(nwritten));
+        }
+      },
+      abort() {
+        core.tryClose(rid);
+      },
+    });
+  }
+
+  function createReadableIOStream(rid) {
+    return new ReadableStream({
+      async pull(controller) {
+        const view = controller.byobRequest.view;
+        const res = await read(rid, view);
+        if (res === null) {
+          core.close(rid);
+          controller.close();
+        } else {
+          controller.byobRequest.respond(res);
+        }
+      },
+      cancel() {
+        core.close(rid);
+      },
+      type: "bytes",
+      autoAllocateChunkSize: 16384,
+    });
+  }
+
+  class Command {
+    #options;
+
+    constructor(command, {
+      args = [],
+      cwd = undefined,
+      clearEnv = false,
+      env = {},
+      gid = undefined,
+      uid = undefined,
+    } = {}) {
+      this.#options = {
+        cmd: pathFromURL(command),
+        args: ArrayPrototypeMap(args, String),
+        cwd: pathFromURL(cwd),
+        clearEnv,
+        env: ObjectEntries(env),
+        gid,
+        uid,
+      };
+    }
+
+    spawn(options = {}) {
+      const child = core.opSync("op_command_spawn", {
+        ...this.#options,
+        ...options,
+      });
+
+      return new Child(illegalConstructorKey, child);
+    }
```

Comment:
I believe these methods should be one-time use. Ie. if you called any of the `spawn`, `status`, `output` the next call should error saying that `Command` has already been "used"

Reply:
this faintly relates to https://github.com/denoland/deno/pull/11618#discussion_r685257615

Reply:
done